staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go

```diff
@@ -110,29 +110,34 @@ func (t *azureDiskCSITranslator) TranslateInTreePVToCSI(pv *v1.PersistentVolume)
 		return nil, fmt.Errorf("pv is nil or Azure Disk source not defined on pv")
 	}
 
-	azureSource := pv.Spec.PersistentVolumeSource.AzureDisk
-
-	// refer to https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/driver-parameters.md
-	csiSource := &v1.CSIPersistentVolumeSource{
-		Driver:           AzureDiskDriverName,
-		VolumeHandle:     azureSource.DataDiskURI,
-		ReadOnly:         *azureSource.ReadOnly,
-		FSType:           *azureSource.FSType,
-		VolumeAttributes: map[string]string{azureDiskKind: "Managed"},
-	}
+	var (
+		azureSource = pv.Spec.PersistentVolumeSource.AzureDisk
+
+		// refer to https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/driver-parameters.md
+		csiSource = &v1.CSIPersistentVolumeSource{
+			Driver:           AzureDiskDriverName,
+			VolumeAttributes: map[string]string{azureDiskKind: "Managed"},
+			VolumeHandle:     azureSource.DataDiskURI,
+		}
+	)
 
 	if azureSource.CachingMode != nil {
 		csiSource.VolumeAttributes[azureDiskCachingMode] = string(*azureSource.CachingMode)
 	}
 
 	if azureSource.FSType != nil {
+		csiSource.FSType = *azureSource.FSType
 		csiSource.VolumeAttributes[azureDiskFSType] = *azureSource.FSType
 	}
 
 	if azureSource.Kind != nil {
 		csiSource.VolumeAttributes[azureDiskKind] = string(*azureSource.Kind)
 	}
 
+	if azureSource.ReadOnly != nil {
+		csiSource.ReadOnly = *azureSource.ReadOnly
+	}
+
 	pv.Spec.PersistentVolumeSource.AzureDisk = nil
 	pv.Spec.PersistentVolumeSource.CSI = csiSource
 	pv.Spec.AccessModes = backwardCompatibleAccessModes(pv.Spec.AccessModes)
```

Comment:
could you remove this line for both azure file and azure disk:
```
pv.Spec.AccessModes = backwardCompatibleAccessModes(pv.Spec.AccessModes)
```
I was wrong, `backwardCompatibleAccessModes` is only for gce driver.
And you could also revert `backwardCompatibleAccessModes` code change, thanks.